From 246947e9cbbf04e2a5373f6dcb0385c064c21b55 Mon Sep 17 00:00:00 2001
From: Megha Dey <megha.dey@linux.intel.com>
Date: Fri, 6 Apr 2018 11:10:55 -0700
Subject: [PATCH] chipsec: Fix Unknown symbol issue while loading chipsec
 module

Define a function valid_mmap_phys_addr_range within linux driver if
kernel version is greater than 4.15.0.

Signed-off-by: Megha Dey <megha.dey@linux.intel.com>
---
 drivers/linux/chipsec_km.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/drivers/linux/chipsec_km.c b/drivers/linux/chipsec_km.c
index d99e053..2121f12 100644
--- a/drivers/linux/chipsec_km.c
+++ b/drivers/linux/chipsec_km.c
@@ -544,6 +544,13 @@ int __weak phys_mem_access_prot_allowed(struct file *file,
         return 1;
 }
 
+#if LINUX_VERSION_CODE >= KERNEL_VERSION(4,15,0)
+int valid_mmap_phys_addr_range(unsigned long pfn, size_t size)
+{
+	return 1;
+}
+#endif
+
 #ifndef ARCH_HAS_VALID_PHYS_ADDR_RANGE
 static inline int valid_phys_addr_range(phys_addr_t addr, size_t count)
 {
-- 
1.9.1

